src/platform/cmx/chal_asm_inc.h

```diff
@@ -0,0 +1,14 @@
+/* Segment selectors for the GDT */
+#define SEL_RPL_USR     0x3
+
+#define SEL_NULL	0x00
+#define SEL_KCSEG       0x08    /* Kernel code selector. */
```

Comment:
This file might be more empty for `cmx`.  It is just supposed to hold the constants required in the assembly.

One important variable is `STK_INFO_SZ`.  Importantly, the kernel assumes that at the top of its kernel stack is the `cos_cpu_local_info`.  I'm not sure how your current PR fits into that.